### PR TITLE
PP-502: Fix for emails/acct log getting truncated.

### PR DESCRIPTION
**Pull request by dilip-krishnan**
Issue: resources_used info in emails/acct log gets truncated

#### Issue-ID

* **[PP-502](https://pbspro.atlassian.net/browse/PP-502)**

#### Problem description
* Resources_used info in emails/acct log can be truncated
#### Cause / Analysis
* The server collects up all of the resources_used information and puts it into a buffer of size RESC_USED_BUF_SIZE(2048 bytes). If we have more than RESC_USED_BUF_SIZE, we truncate the entire resources_used string. The same code is used to create the resources_used string for both the accounting log and the mail (qsub -m option).

#### Solution description
* Replace static buffers with dynamic buffers. The buffers are grown through the use of pbs_strcat().

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by ashwathraop** [comment]
Changes look good. I sign-off.

File: test/tests/functional/pbs_acct_log_trunc.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        # Make sure the server log hasn't been truncated
+        log_match = 'resources_used.foo_str=' + hstr
+        m = self.server.log_match("%s;.*%s.*" % (jid, log_match), regexp=True)
+        self.assertTrue(m)
+
+        # Make sure emails are not truncated
+        mailfile = os.environ['MAIL']
+        if not os.path.isfile(mailfile):
+            self.logger.info("Mail file do not exist or mail not setup.\
```
Comment (bhroam) [inline test/tests/functional/pbs_acct_log_trunc.py]: Can you change this message to "Mail file does not exist or mail is not setup." ?

Comment (dilip-krishnan) [inline test/tests/functional/pbs_acct_log_trunc.py]: Done :)

**Comment by bhroam** [comment]
Works for me.  Please rebase for the final merge.

**Comment by dilip-krishnan** [comment]
Hi Bhroam,
   I have rebase the branch with latest master. 

Regards
Dilip

**Comment by bhroam** [comment]
Looks good.  I merged in the PR.

**Pull request merged**